### PR TITLE
feat(sql): add Snowflake support

### PR DESCRIPTION
**Pull request by alespour**
This PR adds Snowflake support to `sql` package.

### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/sql/snowflake.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+		return nil, err
+	}
+	reader.InitColumnTypes(types)
+
+	return reader, nil
+}
+
+// SnowflakeTranslateColumn translates flux colTypes into their corresponding Snowflake column type
+func SnowflakeColumnTranslateFunc() translationFunc {
+	c := map[string]string{
```
Comment (rockstar) [inline stdlib/sql/snowflake.go:187]: Could you please hoist this value definition up and out of the function itself? It's effectively static, so we should define it up above.

File: stdlib/sql/snowflake.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+		flux.TFloat.String():  "FLOAT",
+		flux.TInt.String():    "NUMBER",
+		flux.TString.String(): "TEXT",
+		flux.TBool.String():   "BOOLEAN",
+		flux.TTime.String():   "TIMESTAMP_LTZ",
+	}
+	return func(f flux.ColType, colName string) (string, error) {
+		s, found := c[f.String()]
```
Comment (rockstar) [inline stdlib/sql/snowflake.go:195]: If we don't use `.String()` here, there's no need to use `.String()` in the above mapping.

File: stdlib/sql/snowflake.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+func (m *SnowflakeRowReader) InitColumnNames(names []string) {
+	m.columnNames = names
+}
+
+func (m *SnowflakeRowReader) InitColumnTypes(types []*sql.ColumnType) {
+	fluxTypes := make([]flux.ColType, len(types))
+	for i := 0; i < len(types); i++ {
+		switch types[i].DatabaseTypeName() {
+		case "FIXED", "NUMBER": // FIXED is reported by Snowflake driver
```
Comment (rockstar) [inline stdlib/sql/snowflake.go:119]: Thanks for this comment.

File: stdlib/sql/snowflake.go (hunk 2 above)
Comment (alespour) [reply stdlib/sql/snowflake.go]: Absolutely. I copied that from other db support and just adjusted that to proper types.

**Pull request merged**